Mathlib/Algebra/Module/Torsion.lean

```diff
@@ -866,3 +863,36 @@ theorem isTorsion_iff_isTorsion_int [AddCommGroup M] :
 #align add_monoid.is_torsion_iff_is_torsion_int AddMonoid.isTorsion_iff_isTorsion_int
 
 end AddMonoid
+
+namespace AddSubgroup
+
+variable (A : Type*) [AddCommGroup A] (n : ℤ)
+
+/-- The additive `n`-torsion subgroup for an integer `n`. -/
+@[reducible]
+def torsionBy : AddSubgroup A :=
+  (Submodule.torsionBy ℤ A n).toAddSubgroup
+
+@[inherit_doc]
+scoped notation:max A"["n"]" => torsionBy A n
+
+lemma torsionBy_negative_eq : A[-n] = A[n] := by
+  ext a
+  simp
+
+variable {A} {n : ℕ}
+
+@[simp]
+lemma nsmul_torsionBy (x : A[n]) : n • x = 0 :=
```

Comment:
You've decided to make A[n] a term not a type, so probably there should be a lemma be about `x \in A[n]` as well as this? And probably it should be called `torsionBy.nsmul` (perhaps one of the two variants should have a `'` at the end)

Reply:
I will add corresponding lemmas for `x ∈ A[n]` (which should be provable by simp). As for the naming, for consistency, should the existing `Submodule.smul_torsionBy` etc be moved into the `torsionBy` as well (in a separate PR)?

Reply:
And does this also apply to `torsionBy_neg` and `moduleZModTorsionBy`, rendering `torsionBy.neg` and `torsionBy.moduleZMod`?

Reply:
I am not the best person to talk to about naming conventions, unfortunately. I'd recommend you ask on the Zulip.